### PR TITLE
feat(strict-ts): apply to common 2

### PR DESCRIPTION
**Pull request by capJavert**
This is the one of the PRs in a series to bring our TypeScript types to comply with [strict](https://www.typescriptlang.org/tsconfig/#strict) config option. This will allow us better type checking during builds and development and allow us to ship with more confidence.

We are doing this as a series of PRs to have easier review and also to incrementally deploy to production.

This one focuses on: 
- commons
- some adjustments to existing code that got broken over during process

### CODE REVIEW

File: src/entity/posts/ArticlePost.ts [hunk 1]
```diff
@@ -9,9 +9,9 @@ export class ArticlePost extends Post {
   @Column({ nullable: true })
   publishedAt?: Date;
 
-  @Column({ type: 'text' })
+  @Column({ type: 'text', nullable: true })
   @Index({ unique: true })
-  url: string;
+  url: string | null;
```
Comment (capJavert) [inline src/entity/posts/ArticlePost.ts:14]: it was always nullable was only missing typeorm option on `Column`

File: src/common/feedGenerator.ts [hunk 2]
```diff
@@ -107,7 +107,11 @@ export const feedToFilters = async (
   userId?: string,
 ): Promise<AnonymousFeedFilters> => {
   const settings = await getExcludedAdvancedSettings(con, feedId);
-  const [tags, excludeSources, memberships] = await Promise.all([
+  const [tags, excludeSources, memberships]: [
```
Comment (rebelchris) [inline src/common/feedGenerator.ts:110]: These are gonna be so useful 🙏

File: src/common/feedGenerator.ts [hunk 3]
```diff
@@ -151,12 +155,15 @@ export const feedToFilters = async (
       }
       return acc;
     },
-    { includeTags: [], blockedTags: [] },
+    { includeTags: [], blockedTags: [] } as {
+      includeTags: string[];
+      blockedTags: string[];
+    },
```
Comment (rebelchris) [inline src/common/feedGenerator.ts:161]: Wouldn't this already work?
```suggestion
    { includeTags: string[], blockedTags: string[] }
```

**Review by rebelchris — APPROVED** [review]
Looks good, non blocking comment/question

File: src/common/feedGenerator.ts (hunk 3 above)
Comment (capJavert) [reply src/common/feedGenerator.ts]: that would be syntax error, its not type its `acc` value of `reduce`, but I moved it to `reduce` generic above. 23504d826da8baaca0c014a0f891afb18d665bce

**Pull request merged**